src/osd/PGLog.cc

```diff
@@ -621,8 +621,8 @@ void PGLog::write_log_and_missing_wo_missing(
   _write_log_and_missing_wo_missing(
     t, km, log, coll, log_oid,
     divergent_priors, eversion_t::max(), eversion_t(), eversion_t(),
-    set<eversion_t>(),
-    set<string>(),
+    nullptr,
```

Comment:
the only caller of `PGLog::_write_log_and_missing_wo_missing()` is `PGLog::write_log_and_missing_wo_missing()`. and the latter always pass it  empty `trimmed` and `trimmed_dups`. can we take this chance to remove them and related logic as well. as they are not used at all.